src/reader/parser.rs

```diff
@@ -146,6 +147,12 @@ impl PullParser {
     /// Checks if this parser ignores the end of stream errors.
     pub fn is_ignoring_end_of_stream(&self) -> bool { self.config.c.ignore_end_of_stream }
 
+    /// Retrieves the Doctype from the document if any
+    #[inline]
+    pub fn doctype(&self) -> &Option<String> {
+        &self.data.doctype
```

Comment:
`&Option` should never be used. `Option<&` is equivalent, and more useful in general.

```suggestion
        self.data.doctype.as_deref()
```